app/ghcup/BrickMain.hs

```diff
@@ -59,70 +74,243 @@ import qualified Data.Text.Lazy.Builder        as B
 import qualified Data.Text.Lazy                as L
 import qualified Graphics.Vty                  as Vty
 import qualified Data.Vector                   as V
-import System.Environment (getExecutablePath)
+import           System.Environment (getExecutablePath)
 import qualified System.Posix.Process          as SPP
+import           Optics.TH (makeLenses, makeLensesFor)
+import           Optics.State (use)
+import           Optics.State.Operators ( (.=), (%=), (<%=))
+import           Optics.Operators ((.~), (^.), (%~))
+import           Optics.Getter (view)
+import           Optics.Lens (Lens', lens, toLensVL)
+
+{- Brick's widget:
+It is a FocusRing over many list's. Each list contains the information for each tool. Each list has an internal name (for Brick's runtime)
+and a label which we can use in rendering. This data-structure helps to reuse Brick.Widget.List and to navegate easily across
+
+Consider this code as private. GenericSectionList should not be used directly as the FocusRing should be align with the Vector containing 
+the elements, otherwise you'd be focusing on a non-existent widget with unknown result (In theory the code is safe unless you have an empty section list). 
+
+- To build a SectionList use the safe constructor sectionList
+- To access sections use the lens provider sectionL and the name of the section you'd like to access
+- You can modify Brick.Widget.List.GenericList within GenericSectionList via sectionL but do not
+  modify the vector length
+
+-}
+
+data GenericSectionList n t e
+    = GenericSectionList
+    { sectionListFocusRing :: FocusRing n                   -- ^ The FocusRing for all sections
+    , sectionListElements :: !(Vector (L.GenericList n t e)) -- ^ A key-value vector
+    , sectionListName :: n                                   -- ^ The section list name
+    }
+
```

Comment:
you can think of this as a list of list with a FocusRing keeping track on which list you are. It is inspire by brick's [List](https://hackage.haskell.org/package/brick-1.10/docs/Brick-Widgets-List.html) API In other words

- `GenericList` is provided by `Brick`
- `GenericSectionList` is a collection of `GenericList` and the [`FocusRing`](https://hackage.haskell.org/package/brick-1.10/docs/Brick-Focus.html) which allow dispach events to each section.